### PR TITLE
PR for changes to SocketUTransport to make it compatible with 1.5.8

### PR DESCRIPTION
**Pull request by debruce**
This PR provides an implementation for L1 transport on a TCP socket to and from a standalone python script using Dispatcher. This was validated using a standalone C++ program exercising publishing and seeing corresponding subscription callbacks.
The heart of the implementation is a single socket connected to the dispatcher, and an unordered_map whose keys are a required UUri and an optional filtering UUri, and whose values shared pointers pointing to a list of listeners for that uuri combination. The test code used to verify this is not a part of this PR.

### CODE REVIEW

File: up_client_socket/cpp/include/TupleOfOptionals.h [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	}
+};
+
+template <typename... Types>
+struct hash<tuple<Types...>> {
+	size_t operator()(const tuple<Types...>& t) const { return hash_value(t); }
+
+private:
+	static void hash_combine(size_t& seed, size_t value) {
+		seed ^= value + 0x9e3779b9 + (seed << 6) + (seed >> 2);
```
Comment (gregmedd) [inline up_client_socket/cpp/include/TupleOfOptionals.h:46]: I'll have to trust you here on this line :sweat_smile:

**Review by gregmedd — APPROVED** [review]
Looks good to me

----

I really don't have any complaints. The socket code all looks correct, locking is where it needs to be, the metaprogramming for the tuple hash looks right. There are a few small cleanup items, but none of them are really a dealbreaker.

I see no reason this could not be merged.

**Comment by gregmedd** [comment]
One more minor comment: you could squash the commits now that you know it builds.

**Review by gregmedd — APPROVED** [review]

File: up_client_socket/cpp/CMakeLists.txt
Comment (agosh01) [inline up_client_socket/cpp/CMakeLists.txt:1]: Remove line 21:
find_package(RapidJSON CONFIG REQUIRED)

and all other changes associated with RapidJSON

File: up_client_socket/cpp/conanfile.txt [hunk 2]
```diff
@@ -1,14 +1,13 @@
 [requires]
+up-core-api/1.6.0
 protobuf/3.21.12
-up-cpp/0.1.2-dev
-rapidjson/cci.20230929
+up-cpp/1.0.1-rc0
```
Comment (agosh01) [inline up_client_socket/cpp/conanfile.txt:4]: Should this be rc1 ?

**Review by agosh01 — APPROVED** [review]
LGTM

**Review by gregmedd — APPROVED** [review]
Gave this another quick looking over. I think it is ready to merge.

File: up_client_socket/cpp/tst.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env python3
+
+from threading import Thread
+# from uprotocol.proto.uattributes_pb2 import UMessageType, UPriority
+# from uprotocol.proto.ustatus_pb2 import UCode
```
Comment (gregmedd) [inline up_client_socket/cpp/tst.py:5]: The linter wants these lines removed
```suggestion
```

File: up_client_socket/cpp/tst.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+from dispatcher.dispatcher import Dispatcher
+
```
Comment (gregmedd) [inline up_client_socket/cpp/tst.py:7]: The linter wants these lines changed:
```suggestion
from dispatcher.dispatcher import Dispatcher
from threading import Thread

```

**Pull request merged**